packages/components/multiselect/src/Multiselect.tsx

```diff
@@ -72,6 +72,11 @@ export interface MultiselectProps extends CommonProps {
    */
   toggleRef?: React.Ref<HTMLButtonElement>;
 
+  /**
+   * Use this prop to get a ref to the reset search button of the component
+   */
+  resetSearchRef?: React.Ref<HTMLButtonElement>;
```

Comment:
Why are you using a ref here instead of passing a callback `onSearchReset` that is called (when defined) in the onClick handler `resetSearchInput ` of the `IconButton`?

Reply:
Resetting the Search Field is not super trivial (its a react 16+ issue, because you can no longer access the native value of a text input https://stackoverflow.com/a/46012210/17269164) , this component already has this function properly implemented and so it would be a waste of time to write the exact same code to reset that field every time you want to have the search input field
https://github.com/contentful/forma-36/blob/main/packages/components/multiselect/src/Multiselect.tsx#L185

So just offering a ref that calls this function is the proper way to go. We also do this already for the toggle menu.

Reply:
Will try to see if i can refactor it to use an imperative ref to only expose the clear functin, thanks :)

Reply:
I do not think this works, because this component also exposes other refs for other parts of it (toggling the list e.g.) and imperativeHandle only works when the component is a forwardRef component. It does not really fit our use-case here unfortunately :/

Reply:
But the MultiSelect is wrapped in a `forwardRef` or what are you referring to?

Reply:
That forwardRef is already used as a reference to the outer popover component. So i can not setup a second forwardRef for that clearSearch Button.

Reply:
What i could do is refactor all refs to be one imperative forwardRef, which would be a breaking change at this point in time.

Reply:
@denkristoffer @mshaaban0 @cf-remylenoir what do you think about the usage of impertative Refs for big components like this one?

So far the component is not that widely used, so its a good idea to do any breaking changes while its still considered alpha?

Reply:
Considering this is an alpha component I reckon it is okay to change its API if necessary 👍  

Ensuring we notify the [consumers](https://github.com/search?q=org%3Acontentful+%3CMultiselect+-repo%3Acontentful%2Fforma-36++NOT+is%3Aarchived&type=code) and follow-up to update the usage accordingly.

Reply:
I spoke with Kristoffer and we both think its better to take a change to the way we approach refs into a bigger picture so that forma components approach refs always the same instead of using an imperative ref approach this one time.

What i will do is to group the props better.